### PR TITLE
adding g1gc and gc log/detail attributes

### PR DESCRIPTION
**Pull request by msilvey**
I would like to be able to use this chef recipe with the G1 GC enabled and/or be able to enable GC logging for all my Cassandra nodes via the template.

### CODE REVIEW

File: templates/default/cassandra-env.sh.erb [hunk 1]
```diff
@@ -273,6 +282,19 @@ if [ "$JVM_VERSION_SHORT" = "1.7" ] && [ "$JVM_ARCH" = "64-Bit" ] ; then
...
 
+<% if node[:cassandra][:jvm] && node[:cassandra][:jvm][:gcdetail] -%>
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCDetails"
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCDateStamps"
+JVM_OPTS="$JVM_OPTS -XX:+PrintHeapAtGC"
+JVM_OPTS="$JVM_OPTS -XX:+PrintTenuringDistribution"
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCApplicationStoppedTime"
+JVM_OPTS="$JVM_OPTS -XX:+PrintPromotionFailure"
+JVM_OPTS="$JVM_OPTS -XX:PrintFLSStatistics=1"
+JVM_OPTS="$JVM_OPTS -Xloggc:/var/log/cassandra/gc.log"
```
Comment (michaelklishin) [inline templates/default/cassandra-env.sh.erb]: I think this should respect Cassandra directory attributes.

File: templates/default/cassandra-env.sh.erb [hunk 2]
```diff
@@ -258,14 +258,23 @@ fi
...
+<% if node[:cassandra][:jvm] && node[:cassandra][:jvm][:g1] -%>
+JVM_OPTS="$JVM_OPTS -XX:+UseG1GC"
+JVM_OPTS="$JVM_OPTS -XX:SurvivorRatio=8"
+JVM_OPTS="$JVM_OPTS -XX:MaxTenuringThreshold=1"
+JVM_OPTS="$JVM_OPTS -XX:+UseTLAB"
+<% else -%>
 JVM_OPTS="$JVM_OPTS -XX:+UseParNewGC"
 JVM_OPTS="$JVM_OPTS -XX:+UseConcMarkSweepGC"
 JVM_OPTS="$JVM_OPTS -XX:+CMSParallelRemarkEnabled"
-JVM_OPTS="$JVM_OPTS -XX:SurvivorRatio=<%= node['cassandra']['gc_survivor_ratio'] %>"
```
Comment (michaelklishin) [inline templates/default/cassandra-env.sh.erb]: Why were these attributes removed in favour of hardcoded values?

**Comment by msilvey** [comment]
I updated for log_dir and added the configurable options back properly.  I did some cut and pasting from the older version we are using and didn't check things properly.

**Comment by michaelklishin** [comment]
@msilvey thank you!

**Pull request merged**